src/manifest/pythonEnvironment.ts

```diff
@@ -118,50 +114,66 @@ export class PythonEnvironment implements Disposable {
-                "Ignoring invalid property  " + prop,
+                "pythonEnvironment:envVars",
+                `Overriding env ${key} from ${envVars[key].from} to ${from}`,
               );
-              continue;
             }
-            const vsCodeEnv = env[prop];
-            const newEnvVars = this.parseEnvVarsFromUserSettings(vsCodeEnv);
-            this.dbtTerminal.debug(
-              "pythonEnvironment",
-              "Loading env vars from config.terminal.integrated.env",
-              "Merging from " + prop,
-              newEnvVars,
-            );
-            envVars = {
-              ...process.env,
-              ...envVars,
-              ...newEnvVars,
-            };
+            envVars[key] = { value: _envVars[key], from: from };
           }
-        }
+        };
+        setEnvVars(process.env, "process");
         try {
+          const integratedEnv:
+            | Record<string, Record<string, string>>
+            | undefined = workspace
+            .getConfiguration("terminal")
+            .get("integrated.env");
+          if (integratedEnv) {
+            // parse vs code environment variables
+            for (const prop in integratedEnv) {
+              // Ignore any settings not supported by the terminal
+              // We don't know which os is used in terminal unfortunately, so we just merge all of them.
+              if (!["osx", "windows", "linux"].includes(prop)) {
+                this.dbtTerminal.debug(
+                  "pythonEnvironment:envVars",
+                  "Loading env vars from config.terminal.integrated.env",
+                  "Ignoring invalid property  " + prop,
+                );
+                continue;
+              }
+              const newEnvVars = parseEnvVarsFromUserSettings(
+                integratedEnv[prop],
+              );
+              this.dbtTerminal.debug(
+                "pythonEnvironment:envVars",
+                "Loading env vars from config.terminal.integrated.env",
+                "Merging from " + prop,
+                Object.keys(newEnvVars),
+              );
+              setEnvVars(newEnvVars, "integrated");
+            }
+          }
           if (api.environment) {
-            envVars = {
-              ...envVars,
-              ...api.environments.getEnvironmentVariables(
-                workspace.workspaceFolders![0],
-              ),
-            };
+            const workspacePath = workspace.workspaceFolders![0];
+            this.dbtTerminal.debug(
+              "pythonEnvironment:envVars",
+              `workspacePath:${workspacePath.uri.fsPath}`,
+            );
+            const workspaceEnv =
+              api.environments.getEnvironmentVariables(workspacePath);
+            this.dbtTerminal.debug(
+              "pythonEnvironment:envVars",
+              `workspaceEnv:${Object.keys(workspaceEnv)}`,
+            );
+            setEnvVars(workspaceEnv, "dotenv");
```

Comment:
Can we move this code out of the python extension piece. it is unrelated to the python extension interface.